### PR TITLE
[Server MP] REST API files only + class/interface implementation

### PR DESCRIPTION
**Pull request by tvallin**
Signed-off-by: tvallin <thibault.vallin@oracle.com>

### CODE REVIEW

**Comment by spericas** [comment]
Didn't you find calling `processSupportingFiles(modifiable, unmodifiable)` useful?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaHelidonServerCodegen.java [hunk 1]
```diff
@@ -168,11 +177,14 @@ public void processOpts() {
         }
 
         if (isLibrary(HELIDON_MP)) {
-            String resourceFolder = "src" + File.separator + "main" + File.separator + "resources";
-            String metaInfFolder = resourceFolder + File.separator + "META-INF";
-            supportingFiles.add(new SupportingFile("RestApplication.mustache", invokerFolder, "RestApplication.java"));
-            supportingFiles.add(new SupportingFile("microprofile-config.properties.mustache", metaInfFolder, "microprofile-config.properties"));
-            supportingFiles.add(new SupportingFile("beans.xml.mustache", metaInfFolder, "beans.xml"));
+            apiTemplateFiles.put("api_impl.mustache", ".java");
```
Comment (aserkes) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaHelidonServerCodegen.java:180]: if you change it to 
```
apiTemplateFiles.put("api_impl.mustache", "Impl.java");
```
you will not need to override `public String apiFilename(String templateName, String tag)`

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaHelidonServerCodegen.java [hunk 2]
```diff
@@ -46,13 +46,15 @@ public class JavaHelidonServerCodegen extends JavaHelidonCommonCodegen {
...
     public static final String INTERFACE_ONLY = "interfaceOnly";
+    public static final String API_FILES_ONLY = "apiFilesOnly";
 
     protected boolean useBeanValidation = true;
     protected String implFolder = "src/main/java";
     protected String serializationLibrary = null;
 
     private boolean interfaceOnly = false;
     private boolean fullProject = false;
+    private boolean apiFilesOnly = false;
```
Comment (aserkes) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaHelidonServerCodegen.java:57]: you can use `fullProject`

**Review by aserkes — CHANGES_REQUESTED** [review]

**Review by aserkes — APPROVED** [review]

**Review by tjquinno — APPROVED** [review]

**Comment by barchetta** [comment]
See https://github.com/oracle/helidon/issues/4708

**Pull request merged**